src/plopp/functions/common.py

```diff
@@ -36,23 +44,50 @@ def _to_data_array(obj):
     return out
 
 
-def _convert_if_not_none(x, unit):
+def _convert_if_not_none(x: Variable, unit: str) -> Union[None, Variable]:
+    """
+    Convert input to the required unit if it is not ``None``.
+    """
     if x is not None:
         return to_unit(number_to_variable(x), unit=unit)
-    return x
 
 
-def _check_size(obj):
+def _check_size(da: DataArray):
+    """
+    Prevent slow figure rendering by raising an error if the data array exceeds a
+    default size.
+    """
     limits = {1: 1_000_000, 2: 2500 * 2500}
-    if obj.ndim not in limits:
+    if da.ndim not in limits:
         raise ValueError("plot can only handle 1d and 2d data.")
-    if prod(obj.shape) > limits[obj.ndim]:
-        raise ValueError(f"Plotting data of size {obj.shape} may take very long or use "
+    if prod(da.shape) > limits[da.ndim]:
+        raise ValueError(f"Plotting data of size {da.shape} may take very long or use "
                          "an excessive amount of memory. This is therefore disabled by "
                          "default. To bypass this check, use `ignore_size=True`.")
 
 
-def preprocess(obj, crop=None, name='', ignore_size=False):
+def preprocess(obj: Union[ndarray, Variable, DataArray],
+               crop: Dict[str, Dict[str, Variable]] = None,
+               name: str = '',
+               ignore_size: bool = False):
+    """
+    Pre-process input data for plotting.
+    This involves:
+      - converting the input to a data array
```

Comment:
reSt requires an empty line before an item list

Didn't check for other places...